plugin/src/test/java/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinitionRJRTest.java

```diff
@@ -21,4 +22,26 @@ private static void doesItSmoke(JenkinsRule r) throws Exception {
         p.setDefinition(new CpsFlowDefinition("print Jenkins.get().getRootDir().toString()", false));
         r.assertBuildStatusSuccess(p.scheduleBuild2(0));
     }
+
+    @Test
+    public void flushLogsOnShutdown() throws Throwable {
+        rjr.then(CpsFlowDefinitionRJRTest::flushLogsOnShutdownPreRestart);
+        rjr.then(CpsFlowDefinitionRJRTest::flushLogsOnShutdownPostRestart);
+    }
+
+    private static void flushLogsOnShutdownPreRestart(JenkinsRule r) throws Exception {
+        WorkflowJob p = r.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition("sleep 10\n", true));
+        WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+        r.waitForMessage("Sleeping for 10 sec", b);
+        r.jenkins.doQuietDown();
+    }
+
+    private static void flushLogsOnShutdownPostRestart(JenkinsRule r) throws Exception {
+        WorkflowJob p = r.jenkins.getItemByFullName("p", WorkflowJob.class);
+        WorkflowRun b = p.getLastBuild();
+        r.assertBuildStatusSuccess(r.waitForCompletion(b));
+        r.assertLogContains("Resuming build at ", b);
+        r.assertLogContains("Pausing (Preparing for shutdown)", b);
```

Comment:
The first assertion here passes with or without the fix, since that message is printed after the restart. However, the second assertion is printed right before Jenkins shuts down, and it is missing from the log if you revert the fix.